src/gt4py/stencil_object.py

```diff
@@ -267,6 +267,15 @@ def _validate_args(self, field_args, param_args, domain, origin) -> None:
                         f"expects {field_domain_ndim + len(field_info.data_dims)} ('{field_info.axes}[{field_info.data_dims}]')"
                     )
 
+                # Check: data dimensions shape
+                if not all(
+                    field.shape[field_domain_ndim + i] == dim_size
+                    for i, dim_size in enumerate(field_info.data_dims)
+                ):
```

Comment:
Maybe this check can be simplified like this:
```suggestion
                if field.shape[field_domain_ndim:] != field_info.data_dims:
```
Or am I missing something? Can `field.shape` have more dimensions?

Reply:
No it cannot as far as I know. This is a nice simplification!